File: SkinApi/Models/SkinShard.cs
using System.Text.Json.Serialization;
using SkinApi.Enums;

namespace SkinApi.Models;

public class SkinShard
{
    public int Id { get; set; }
    public string Name { get; set; }
    public RarityEnum Rarity { get; set; }
    public bool Legacy { get; set; }
    public int DisenchantPrice { get; set; }
    public int Price { get; set; }
    public int ChampionId { get; set; }
    [JsonIgnore]
    public Champion Champion { get; set; }
}

File: front-end/src/types/skinShard.ts
import type { Rarity } from "./champion";

export interface SkinShardHasSkin {
    id: number;
    name: string;
    champion: string;
    championId: number;
    rarity: Rarity;
    legacy: boolean;
    disenchantPrice: number;
    price: number;
    hasSkin: boolean;
    bestSkinRarity: Rarity;
}

export interface CreateSkinShard {
    name: string;
    championId: number;
    rarity: number;
    legacy: boolean;
    disenchantPrice: number;
    price: number;
}